emails/views.py

```diff
@@ -118,6 +123,159 @@ def _index_DELETE(request_data, user_profile):
     return redirect('profile')
 
 
+@csrf_exempt
+def sns_inbound(request):
+    # We can check for some invalid values in headers before processing body
+    topic_arn = request.headers.get('X-Amz-Sns-Topic-Arn', None)
+    if not topic_arn:
+        logger.error('SNS inbound request without X-Amz-Sns-Topic-Arn')
+        return HttpResponse(
+            'Received SNS request without Topic ARN.', status=400
+        )
+    if topic_arn != settings.AWS_SNS_TOPIC:
+        logger.error(
+            'SNS message for wrong ARN',
+            extra={
+                'configured_arn': settings.AWS_SNS_TOPIC,
+                'received_arn': topic_arn,
+            }
+        )
+        return HttpResponse(
+            'Received SNS message for wrong topic.', status=400
+        )
+
+    message_type = request.headers.get('X-Amz-Sns-Message-Type', None)
+    if not message_type:
+        logger.error('SNS inbound request without X-Amz-Sns-Message-Type')
+        return HttpResponse(
+            'Received SNS request without Message Type.', status=400
+        )
+    if message_type not in SUPPORTED_SNS_TYPES:
+        logger.error(
+            'SNS message for unsupported type',
+            extra={
+                'supported_sns_types': SUPPORTED_SNS_TYPES,
+                'message_type': message_type,
+            }
+        )
+        return HttpResponse(
+            'Received SNS message for unsupported Type: %s' % message_type,
+            status=400
+        )
+
+    json_body = json.loads(request.body)
+    try:
+        verified_json_body = verify_from_sns(json_body)
+    except Exception:
+        logger.error(
+            'SNS message with invalid signature',
+            extra={
+                'SigningCertURL': json_body['SigningCertURL'],
+                'Signature': json_body['Signature'],
+            }
+        )
+        return HttpResponse(
+            'Received SNS message with invalid signature: %s' % message_type,
+            status=401
+        )
+
+    return _sns_inbound_logic(topic_arn, message_type, verified_json_body)
+
+
+def _sns_inbound_logic(topic_arn, message_type, json_body):
+    if message_type == 'SubscriptionConfirmation':
+        logger.info(
+            'SNS SubscriptionConfirmation',
+            extra={'SubscribeURL': json_body['SubscribeURL']}
+        )
+        return HttpResponse('Logged SubscribeURL', status=200)
+    if message_type == 'Notification':
+        logger.info(
+            'SNS Notification',
+            extra={'json_body': json_body},
+        )
+        return _sns_notification(json_body)
+
```

Comment:
```suggestion
        return _sns_notification(json_body)
    logger.error(
        'SNS message type did not fall under the SNS inbound logic',
        extra={'message_type': message_type}
    )
    return HttpResponse('Received SNS message with type not handled in inbound log', status=400)
```
We should have a default behavior.

Reply:
Gonna merge early to test dev env. Can you file an issue for this?

Reply:
Filed #364 for this.